src/main/java/org/dependencytrack/persistence/QueryManager.java

```diff
@@ -189,6 +192,33 @@ public QueryManager(final PersistenceManager pm, final AlpineRequest request) {
         this.request = request;
     }
 
+    @Override
+    public Query decorate(final Query query) {
+        // Clear the result to fetch if previously specified (i.e. by getting count)
+        query.setResult(null);
+        if (pagination != null && pagination.isPaginated()) {
+            final long begin = pagination.getOffset();
+            final long end = begin + pagination.getLimit();
+            query.setRange(begin, end);
+        }
+        if (orderBy != null && RegexSequence.Pattern.STRING_IDENTIFIER.matcher(orderBy).matches() && orderDirection != OrderDirection.UNSPECIFIED) {
+            // Check to see if the specified orderBy field is defined in the class being queried.
+            boolean found = false;
+            final org.datanucleus.store.query.Query iq = ((JDOQuery) query).getInternalQuery();
+            final String candidateField = orderBy.contains(".") ? orderBy.substring(0, orderBy.indexOf('.')) : orderBy;
+            for (final Field field: iq.getCandidateClass().getDeclaredFields()) {
+                if (candidateField.equals(field.getName())) {
+                    found = true;
+                    break;
+                }
+            }
+            if (found) {
+                query.setOrdering(orderBy + " " + orderDirection.name().toLowerCase() + " " + "id asc");
```

Comment:
I think there is a comma missing here:

```suggestion
                query.setOrdering(orderBy + " " + orderDirection.name().toLowerCase() + ", id asc");
```

Reply:
It was! Interestingly, there was no exception, just ordering part was truncated when it was added without comma